### PR TITLE
EFAX-28 implemented pdf flattening service

### PR DESCRIPTION
**Pull request by KDNeufeld**
# Description

This PR includes the following proposed change(s):

[EFAX-28](https://justice.gov.bc.ca/jira/browse/EFAX-28)

- removed old, legacy outputservice in favour of a new flattening service.
- added xsd of new flattening service
- enhanced pom.xml to automatically generated model objects for service
- implemented the webServiceTemplate that calls this new service.
- added junit tests
- removed PDF_USERNAME and PDF_PASSWORD since new service doesn't need them.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

mvn clean compile test

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [x] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/main/java/ca/bc/gov/ag/efax/pdf/config/PdfConfiguration.java [hunk 1]
```diff
@@ -3,20 +3,18 @@
...
-				pdfProperties.getEndpoint(), 
-				pdfProperties.getUsername(), 
-				pdfProperties.getPassword());
-		return outputServiceUtils;
-	}
-	
+    @Bean
+    public Jaxb2Marshaller pdfMarshaller() {
+        Jaxb2Marshaller marshaller = new Jaxb2Marshaller();
+        // this package must match the package in the <generatePackage> specified in pom.xml
```
Comment (akroon3r) [inline src/main/java/ca/bc/gov/ag/efax/pdf/config/PdfConfiguration.java:15]: what happens if it doesn't match?

File: src/main/java/ca/bc/gov/ag/efax/pdf/service/PdfService.java [hunk 2]
```diff
@@ -1,64 +1,82 @@
...
+        setDefaultUri(pdfProperties.getEndpoint());
+        setMarshaller(pdfMarshaller);
+        setUnmarshaller(pdfMarshaller);
+    }
 
-    public File flattenPdf(String urlString, String fileName) {
-        FileOutputStream fos = null;
+    /**
+     * Attempts to flatten (aka normalize/simplify) a PDF by delegating the call to a web service.
+     * @param url a URL to a PDF to flatten
```
Comment (akroon3r) [inline src/main/java/ca/bc/gov/ag/efax/pdf/service/PdfService.java:41]: isn't this the api gateway endpoint?

File: src/main/resources/PdfTransformations.xsd [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+		</xsd:sequence>
+	</xsd:complexType>
+	<xsd:complexType name="PDFTransformationsResponse">
+		<xsd:sequence>
+			<xsd:element name="PDFTransformationsReturn" type="xsd:base64Binary" />
+		</xsd:sequence>
+	</xsd:complexType>
+	<xsd:element name="PDFTransformations" type="tns:PDFTransformations" />
+	<xsd:element name="PDFTransformationsResponse" type="tns:PDFTransformationsResponse" />
+</xsd:schema>
```
Comment (akroon3r) [inline src/main/resources/PdfTransformations.xsd:18]:
```suggestion
</xsd:schema>

```

File: src/main/java/ca/bc/gov/ag/efax/pdf/config/PdfConfiguration.java (hunk 1 above)
Comment (KDNeufeld) [reply src/main/java/ca/bc/gov/ag/efax/pdf/config/PdfConfiguration.java:15]: The package locations need to match (of where the model is stored and how it used) or the marshaller won't know where the model is when serializing/deserializing objects from the remote webservice.

**Review by akroon3r — COMMENTED** [review]
Just a couple of dumb questions.

File: src/main/java/ca/bc/gov/ag/efax/pdf/service/PdfService.java (hunk 2 above)
Comment (KDNeufeld) [reply src/main/java/ca/bc/gov/ag/efax/pdf/service/PdfService.java:41]: Not quite.  When JUSTIN sends a fax, the attachment is a URL (not an included attachment) to a PDF doc.  This method will take this file attachment (URL) to a pdf and attempt to flatten it.

**Review by akroon3r — APPROVED** [review]
Thanks :)

**Comment by KDNeufeld** [comment]
:) No question is a dumb question - not ever.
Thanks for the review!

**Pull request merged**